src/hooks/useWindowWidth.js

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from 'react';
+
+const useWindowWidth = () => {
+  const [width, setWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleResize = () => setWidth(window.innerWidth);
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  });
```

Comment:
```suggestion
  useEffect(() => {
    const handleResize = () => setWidth(window.innerWidth);
    window.addEventListener('resize', handleResize);

    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []); 
```

It would be better if adding it, since event listener no need to re-bind every time.
It only runs at component mount, and cancel when component unmount.

Reply:
Got it, thanks